pkg/controller/flinkapplication/controller.go

```diff
@@ -36,6 +38,7 @@ type ReconcileFlinkApplication struct {
 	cache             cache.Cache
 	metrics           *reconcilerMetrics
 	flinkStateMachine FlinkHandlerInterface
+	locks             *SyncMap
```

Comment:
You can may just use [`m  sync.Map`](https://pkg.go.dev/sync#Map) to synchronize state:

```suggestion
```

Reply:
Go team has fixed many of them, but there are a couple of rare cases when built-in implementation is slow.

Please see https://github.com/golang/go/issues/28938